src/Command/EnumCommand.php

```diff
@@ -0,0 +1,99 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         3.1.0
+ * @license       https://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Bake\Command;
+
+use Cake\Console\Arguments;
+use Cake\Console\ConsoleOptionParser;
+use InvalidArgumentException;
+
+/**
+ * Enum code generator.
+ */
+class EnumCommand extends SimpleBakeCommand
+{
+    /**
+     * Task name used in path generation.
+     *
+     * @var string
+     */
+    public string $pathFragment = 'Model/Enum/';
+
+    /**
+     * @inheritDoc
+     */
+    public function name(): string
+    {
+        return 'enum';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function fileName(string $name): string
+    {
+        return $name . '.php';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function template(): string
+    {
+        return 'Bake.Model/enum';
+    }
+
+    /**
+     * Get template data.
+     *
+     * @param \Cake\Console\Arguments $arguments The arguments for the command
+     * @return array
+     * @phpstan-return array<string, mixed>
+     */
+    public function templateData(Arguments $arguments): array
+    {
+        $data = parent::templateData($arguments);
+
+        $backed = $arguments->getOption('backed');
+        if ($backed && !in_array($backed, ['string', 'int'], true)) {
+            throw new InvalidArgumentException('Backed enums must be of type `string` or `int`');
+        }
+
+        $data['backed'] = $backed;
+
+        return $data;
+    }
+
+    /**
+     * Gets the option parser instance and configures it.
+     *
+     * @param \Cake\Console\ConsoleOptionParser $parser The option parser to update.
+     * @return \Cake\Console\ConsoleOptionParser
+     */
+    public function buildOptionParser(ConsoleOptionParser $parser): ConsoleOptionParser
+    {
+        $parser = $this->_setCommonOptions($parser);
+
+        $parser->setDescription(
+            'Bake (backed) enums for use in models.'
+        )->addOption('backed', [
```

Comment:
Think this should be `backing` no` backed` since it will always be backed.

At least, that's what makes sense to me in english. Maybe it's harder for non-english speakers to understand the difference though.

Reply:
I agree, hence I suggested `backing-type` earlier or shorter `backing` works too.

Reply:
`backing-type` is clearer.

Reply:
Wait, but without that Option the enum is not backed. So for me the original Name still seems to make sense. 
I am fine changing it to your suggestion if u think it makes more sense.

Reply:
> Wait, but without that Option the enum is not backed

Why do we need to bake non-backed enums? The db layer only supports backed enums, so it's only a question of deciding whether to have `int` return type or `string`, defaulting to the latter.

Reply:
True, the db layer sure does
But maybe people want to use enums in their (model) layer or otherwise and those wouldnt necessarily need to be backed.
So I thought:
- normal (default)
- backing-type int or string

Reply:
Then for such enums I don't see why such enums would need a `label()` method too and without that such classes can be easily created by hand.

I have opened https://github.com/cakephp/bake/pull/970 which my suggested changes on top of this PR.

Reply:
Making `string` backed enums would be the most common use case and having to specify `-b string` each time would be an unnecessary chore.